agent/agentcontainers/api.go

```diff
@@ -1173,6 +1174,14 @@ func (api *API) maybeInjectSubAgentIntoContainerLocked(ctx context.Context, dc c
 
 				appsWithPossibleDuplicates = append(appsWithPossibleDuplicates, customization.Apps...)
 			}
+
+			// NOTE(DanielleMaywood):
+			// We only want to take an agent name specified in the very last customization layer.
+			// This restricts the ability for a feature to specify the agent name. We may revisit
+			// this in the future, but for now we want to restrict this behavior.
+			if len(coderCustomization) > 0 {
+				possibleAgentName = coderCustomization[len(coderCustomization)-1].Name
```

Comment:
If this is gibberish/invalid, we will simply fail agent creation. It'd be best to validate it before considering it (and log + rely on fallbacks if bad).